pkg/utils/errors/errors.go

```diff
@@ -1,9 +1,10 @@
 package errors
 
 import (
```

Comment:
Package `pkg/utils/errors` is used only in `agent/main.go` which only demonstrates how to use gitops-engine but it used 400+ times in Argo CD. 

I think it indicates that `pkg/utils/errors` don't belong to `gitops-engine` at all .

Can you please remove `pkg/utils/errors` completely and stop using it  `agent/main.go`?